test/e2e/cross_os/test_job_submissions.py

```diff
@@ -632,47 +630,195 @@ def test_worker_uses_job_attachment_configuration(
+            assert output_file_content == (input_file_content + test_run_uuid)
+            os.remove(os.path.join(list(output_path.keys())[0], "output_file"))
 
-            # Set root path output will be downloaded to to output_root_path. Assumes there is only one root path.
-            job_output_downloader.set_root_path(
-                list(output_paths_by_root.keys())[0],
-                tmp_dir_name,
+    @pytest.mark.parametrize(
+        "hash_string_script",
+        [
+            (
+                '#!/usr/bin/env bash\n\nfolder_path={{Param.DataDir}}/files\ncombined_contents=""\nfor file in "$folder_path"/*; do\n   if [ -f "$file" ]; then\n      combined_contents+="$(cat "$file" | tr -d \'\\n\')"\n   fi\ndone\nsha256_hash=$(echo -n "$combined_contents" | sha256sum | awk \'{ print $1 }\')\necho -n "$sha256_hash" > {{Param.DataDir}}/output_file.txt'
+                if os.environ["OPERATING_SYSTEM"] == "linux"
+                else '$InputFolder = "{{Param.DataDir}}\\files"\n$OutputFile = "{{Param.DataDir}}\\output_file.txt"\n$combinedContent = ""\n$files = Get-ChildItem -Path $InputFolder -File\nforeach ($file in $files) {\n    $combinedContent += [IO.File]::ReadAllText($file.FullName)\n}\n$sha256 = [System.Security.Cryptography.SHA256]::Create().ComputeHash([System.Text.Encoding]::UTF8.GetBytes($combinedContent))\n$hashString = [System.BitConverter]::ToString($sha256).Replace("-", "").ToLower()\nSet-Content -Path $OutputFile -Value $hashString -NoNewLine'
             )
-            job_output_downloader.download_job_output()
-
-            with (
-                open(os.path.join(job_bundle_path, "files", "test_input_file"), "r") as input_file,
-                open(
-                    os.path.join(
-                        tmp_dir_name,
-                        "output_file",
-                    ),
-                    "r",
-                    encoding="utf-8-sig",
-                ) as output_file,
-            ):
-                input_file_content: str = input_file.read()
-                output_file_content = output_file.read()
-
-                # Verify that the output file content is the input file content plus the uuid we appended in the job
-                assert output_file_content == (input_file_content + test_run_uuid)
+        ],
+    )
+    def test_worker_uses_job_attachment_sync(
+        self,
+        deadline_resources: DeadlineResources,
+        deadline_client: DeadlineClient,
+        hash_string_script: str,
+    ) -> None:
+        # Verify that the worker sync job attachment correctly and report the progress correctly as well
+
+        job_bundle_path: str = os.path.join(
+            os.path.dirname(__file__),
+            "job_attachment_bundle_large",
+        )
+        file_path: str = os.path.join(job_bundle_path, "files")
+
+        os.mkdir(job_bundle_path)
+        os.mkdir(file_path)
+
+        # Create 2500 very small files to transfer
+        for i in range(2500):
```

Comment:
Where are you deleting these files?

Reply:
Nvm I see it at the bottom of the test, but that will not always run, if the test fails

Reply:
Maybe better to have a fixture to create these files and cleanup?

Reply:
Since these file is changed depend on each test req, I don't think having it in a fixture is needed.

Reply:
I'd suggest using this: https://docs.pytest.org/en/stable/how-to/tmp_path.html